components/filedata-table.vue

```diff
@@ -29,21 +62,37 @@
 				let tableRow = document.getElementById(itemID);
 
 				// 1. Determine if this item is to be colored or uncolored
-				if ( this.defaultPalette.bColor == tableRow.style.backgroundColor ||
-					 "" == tableRow.style.backgroundColor ) {
-					tableRow.style.backgroundColor = this.currentPalette.bColor;
-					tableRow.style.color = this.currentPalette.fColor;
+				if ( tableRow.classList.contains(this.defaultStyleClass) ) {
+
+					// A. Replace the default style class name with the desired new style
+					tableRow.classList.replace(this.defaultStyleClass, this.currentStyleClass);
 				} else {
-					tableRow.style.backgroundColor = this.defaultPalette.bColor;
-					tableRow.style.color = this.defaultPalette.fColor;
+
+					// A. If a category style class exists, replace it with the desired new style
+					let oldStyleClass = "";
+					for ( let index = 0; index < tableRow.classList.length; index++ ) {
+
+						if ( -1 != tableRow.classList[index].indexOf(this.$store.getters.cssStylePrefix) ) {
+
+							oldStyleClass = tableRow.classList[index];
+							break;
+						}
+					}
+
+					// B. Determine if category link is being replaced or column is being unlinked
+					if ( oldStyleClass == this.currentStyleClass ) {
+						tableRow.classList.replace(oldStyleClass, this.defaultStyleClass);
```

Comment:
This method seems to both handle the updating of column-category mapping (via the emit below) and the updating of the styling (via direct changes to the style classes applied to each row). I think we should keep these two parts separate.  

We can rename this method to something descriptive like `assignCategory` and only handle the updating of the column-to-category mapping. Then a separate `updateTableStyle` (or something like that) method can react to this change by updating the styling of the table.